File: app/src/main/java/com/elitec/alejotaller/feature/agent/domain/entity/AgentMessage.kt
package com.elitec.alejotaller.feature.agent.domain.entity

enum class AgentRole {
    USER,
    ASSISTANT,
    SYSTEM,
    TOOL,
}

enum class AgentConnectionStatus {
    UNKNOWN,
    OK,
    ERROR,
    UNCONFIGURED,
}

data class AgentMessage(
    val id: String,
    val role: AgentRole,
    val content: String,
    val createdAtIso: String,
)

data class AgentConnectionResult(
    val status: AgentConnectionStatus,
    val modelId: String?,
    val message: String,
)

data class AgentReply(
    val message: AgentMessage,
    val providerId: String?,
)

File: web/src/core/feature/agent/domain/entity/AgentMessage.ts
/** Domain — Agent conversation message (Fase 1: texto; tools en Fase 2). */

export type AgentRole = "user" | "assistant" | "system" | "tool";

export type AgentConnectionStatus = "unknown" | "ok" | "error" | "unconfigured";

export interface AgentMessage {
  id: string;
  role: AgentRole;
  content: string;
  createdAt: string;
}

export interface AgentConnectionResult {
  status: AgentConnectionStatus;
  /** Model or agent id used for the probe */
  modelId: string | null;
  message: string;
}

export interface AgentReply {
  message: AgentMessage;
  /** Provider raw id if any */
  providerId: string | null;
}
